src/uri.cr

```diff
@@ -19,6 +20,31 @@ require "./uri/uri_parser"
 # uri.query  # => "id=30&limit=5"
 # uri.to_s   # => "http://foo.com/posts?id=30&limit=5#time=1305298413"
 # ```
+#
+# # URL Encoding
+#
+# This class provides a number of methods for encoding and decoding strings using
+# URL Encoding (also known as Percent Encoding) as defined in [RFC 3986](https://www.ietf.org/rfc/rfc3986.txt)
+# as well as [`x-www-form-urlencoded`](https://url.spec.whatwg.org/#urlencoded-serializing).
+#
+# Each method has two variants, one returns a string, the other writes directly
+# to an IO.
+#
+# * `.decode(string : String, *, plus_to_space : Bool = false) : String`: Decodes a URL-encoded string.
+# * `.decode(string : String, io : IO, *, plus_to_space : Bool = false) : Nil`: Decodes a URL-encoded string to an IO.
+# * `.encode(string : String, *, space_to_plus : Bool = false) : String`: URL-encodes a string.
+# * `.encode(string : String, io : IO, *, space_to_plus : Bool = false) : Nil`: URL-encodes a string to an IO.
+# * `.decode_www_form(string : String, *, plus_to_space : Bool = true) : String`: Decodes an `x-www-form-urlencoded` string component.
+# * `.decode_www_form(string : String, io : IO, *, plus_to_space : Bool = true) : Nil`: Decodes an `x-www-form-urlencoded` string component to an IO.
+# * `.encode_www_form(string : String, *, space_to_plus : Bool = true) : String`: Encodes a string as a `x-www-form-urlencoded` component.
+# * `.encode_www_form(string : String, io : IO, *, space_to_plus : Bool = true) : Nil`: Encodes a string as a `x-www-form-urlencoded` component to an IO.
+#
+# The main difference is that `.encode_www_form` encodes reserved characters
+# (see `.reserved?`), while `.encode` does not. The decode methods are exactly
+# similar except for the handling of `+` characters.
```

Comment:
_exactly similar_? 🤔 ;)

Reply:
Yes, that's the only difference. `decode` decodes `+` literally while `decode_www_form` decodes it as space.

Reply:
I'd rephrase it into sth like `The decode methods are almost identical, except for...`.

Reply:
Okay, I get what you mean. I'm not a native speaker but AFAIK "exactly similar" is a common phrase to describe something that is almost identical. But, sure, I can rephrase that.

Reply:
I don't think this matters much.

Reply:
@asterite yes it does, this wording is confusing and docs are meant to explain things and not obscure, don't they?

Reply:
I don't think it's confusing, but the alternative is also good, so I'll change it.